diesel/src/query_dsl/locking_dsl.rs

```diff
@@ -30,3 +30,63 @@ where
         self.as_query().for_update()
     }
 }
+
+/// The `skip_locked` method
+///
+/// This trait should not be relied on directly by most apps. Its behavior is
+/// provided by [`QueryDsl`]. However, you may need a where clause on this trait
+/// to call `skip_locked` from generic code.
+///
+/// [`QueryDsl`]: ../trait.QueryDsl.html
+pub trait SkipLockedDsl {
+    /// The type returned by `skip_locked`. See [`dsl::SkipLocked`] for
+    /// convenient access to this type.
+    ///
+    /// [`dsl::SkipLocked`]: ../../dsl/type.SkipLocked.html
+    type Output;
+
+    /// See the trait level documentation
+    fn skip_locked(self) -> Self::Output;
+}
+
+impl<T> SkipLockedDsl for T
```

Comment:
We can drop this impl. The only types which implement this already have a `FOR UPDATE` clause, which tables do not.